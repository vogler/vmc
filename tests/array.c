int[5] a;

int main(){
  a[0] = 2;
  a[1] = *a + 3;
  a[2] = *(a+1) + 4;
  return a[2];
}
