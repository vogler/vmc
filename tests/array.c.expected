enter 9
alloc 6
mark
loadc _main
call
slide 5
halt
_main:
enter 4
alloc 0
loadc 2
loadc 1
loadc 0
loadc 1
mul
add
store
pop
loadc 1
load
loadc 3
add
loadc 1
loadc 1
loadc 1
mul
add
store
pop
loadc 1
loadc 1
add
load
loadc 4
add
loadc 1
loadc 2
loadc 1
mul
add
store
pop
loadc 1
loadc 2
loadc 1
mul
add
load
storer -3
return
return
